src/MassTransit.Abstractions/Util/ExceptionUtil.cs

```diff
@@ -39,7 +39,7 @@ public static string GetStackTrace(Exception? exception)
             return _cleanup.Replace(stackTrace, "");
         }
 
-        public static IDictionary<string, object> GetExceptionHeaderDictionary(Exception exception)
+        public static Dictionary<string, object> GetExceptionHeaderDictionary(Exception exception)
```

Comment:
This is changed because the Azure SDK `DeadLetterMessageAsync` overloads for unknown reasons [take different parameter types](https://github.com/Azure/azure-sdk-for-net/issues/43498).